gobblin-yarn/src/main/java/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -0,0 +1,598 @@
+        addFileAsLocalResource(destFilePath, LocalResourceType.FILE, resourceMap.get());
+      }
+    }
+  }
+
+  private void addAppRemoteFiles(String hdfsFileList, Map<String, LocalResource> resourceMap)
+      throws IOException {
+    for (String hdfsFilePath : SPLITTER.split(hdfsFileList)) {
+      addFileAsLocalResource(new Path(hdfsFilePath), LocalResourceType.FILE, resourceMap);
+    }
+  }
+
+  private void addJobConfPackage(String jobConfPackagePath, Path destDir, Map<String, LocalResource> resourceMap)
+      throws IOException {
+    Path srcFilePath = new Path(jobConfPackagePath);
+    Path destFilePath = new Path(destDir, srcFilePath.getName());
+    this.fs.copyFromLocalFile(srcFilePath, destFilePath);
+    addFileAsLocalResource(destFilePath, LocalResourceType.ARCHIVE, resourceMap);
+  }
+
+  private void addFileAsLocalResource(Path destFilePath, LocalResourceType resourceType,
+      Map<String, LocalResource> resourceMap) throws IOException {
+    LocalResource fileResource = Records.newRecord(LocalResource.class);
+    FileStatus fileStatus = this.fs.getFileStatus(destFilePath);
+    fileResource.setResource(ConverterUtils.getYarnUrlFromPath(destFilePath));
+    fileResource.setSize(fileStatus.getLen());
+    fileResource.setTimestamp(fileStatus.getModificationTime());
+    fileResource.setType(resourceType);
+    fileResource.setVisibility(LocalResourceVisibility.APPLICATION);
+    LOGGER.debug(String.format("Created a LocalResource for file %s of type %s",
+        fileResource.getResource(), resourceType));
+    resourceMap.put(destFilePath.getName(), fileResource);
+  }
+
+  private Map<String, String> getEnvironmentVariables() {
+    Map<String, String> environmentVariableMap = Maps.newHashMap();
+
+    Apps.addToEnvironment(environmentVariableMap, ApplicationConstants.Environment.JAVA_HOME.key(),
+        System.getenv(ApplicationConstants.Environment.JAVA_HOME.key()));
+
+    // Add jars/files in the working directory of the ApplicationMaster to the CLASSPATH
+    Apps.addToEnvironment(environmentVariableMap, ApplicationConstants.Environment.CLASSPATH.key(),
+        ApplicationConstants.Environment.PWD.$());
+    Apps.addToEnvironment(environmentVariableMap, ApplicationConstants.Environment.CLASSPATH.key(),
+        ApplicationConstants.Environment.PWD.$() + File.separator + "*");
+
+    String[] classpaths = this.yarnConfiguration.getStrings(YarnConfiguration.YARN_APPLICATION_CLASSPATH,
+        YarnConfiguration.DEFAULT_YARN_APPLICATION_CLASSPATH);
+    if (classpaths != null) {
+      for (String classpath : classpaths) {
+        Apps.addToEnvironment(
+            environmentVariableMap, ApplicationConstants.Environment.CLASSPATH.key(), classpath.trim());
+      }
+    }
+
+    return environmentVariableMap;
+  }
+
+  private String buildApplicationMasterCommand(int memoryMbs) {
+    String appMasterClassName = GobblinApplicationMaster.class.getSimpleName();
+    return String.format(
```

Comment:
I think using `StringBuilder` rather than `String.format` will make the code more readable.